explorer/explorer.go

```diff
@@ -109,27 +107,51 @@ func GetTicketPoolData(interval dbtypes.ChartGrouping) (barGraphs []*dbtypes.Poo
 	return
 }
 
-// cleanUpTicketPoolData provides a thread-safe way to clean up/invalidate the ticketspool
-// cache data. This function should be called when a new block has been added.
-func cleanUpTicketPoolData() {
-	ticketPoolGraphsCache.Lock()
-	defer ticketPoolGraphsCache.Unlock()
+// CleanUpTicketPoolData provides a thread-safe way to clean up/invalidate the
+// ticketpool cache data. It resets the ticketpool cache data and sets the
+// current block height. Check for invalid data every cleanUpTickerInterval.
+func (exp *explorerUI) CleanUpTicketPoolData(quit chan int) {
+	ticker := time.NewTicker(time.Duration(cleanUpTickerInterval) * time.Second)
+	for {
+		select {
+		case <-ticker.C:
+			height := exp.Height()
```

Comment:
Two other options to consider instead of polling as you are doing with this solution:

1. `explorer.GetTicketPoolData` could accept a height parameter, doing the clean up and responding with false when it does not have the data.
2. You can remove the channel from `CleanUpTicketPoolData` and call `CleanUpTicketPoolData` directly from `(*appContext).StatusNtfnHandler` in response to signals on either `UpdateStatusNodeHeight` or `UpdateStatusDBHeight`.